src/main/java/frc/robot/lib/RoboLionsPID.java

```diff
@@ -4,46 +4,43 @@ public class RoboLionsPID {
   //PID Template Code
   //23MARCH2019 Dustin Krack and Ava Byrd
   
-  public double proportionalGain = 0;
-  public double integralGain = 0;
-  public double derivativeGain = 0;
-  public double integral_charge = 0;
+  public double proportionalGain = 0.0;
+  public double integralGain = 0.0;
+  public double derivativeGain = 0.0;
+  public double integral_charge = 0.0;
 
   //errors outside of the zone result of the cage
   //do not allow the integrator to run
-  public double upperCageLimit = 0;
-  public double lowerCageLimit = 0;
+  public double upperCageLimit = 0.0;
+  public double lowerCageLimit = 0.0;
 
-  public double upperDeadbandLimit = 0;
-  public double lowerDeadbandLimit = 0;
+  public double upperDeadbandLimit = 0.0;
+  public double lowerDeadbandLimit = 0.0;
 
-  public double maxOutput = 0;
-  public double minOutput = 0;
+  public double maxOutput = 0.0;
+  public double minOutput = 0.0;
 
-  public double output = 0;
-  //public double input = 0;
-  public double error = 0;
+  public double output = 0.0;
+  public double error = 0.0;
 
-  public double derivativeCalculation = 0;
+  public double derivativeCalculation = 0.0;
 
-  public double previousError = 0;
+  public double previousError = 0.0;
 
   public double deltaTime = 0.02;
 
-  public double cmd = 0;
-  public double feed = 0;
+  public double cmd = 0.0;
+  public double feed = 0.0;
 
   public boolean deadband_active = false;
 
   // These are used to turn off the limiter cage and the deadband 2/15/20
   public boolean enableCage = true;
   public boolean enableDeadBand = true;
 
-  public static int deadband_counter = 0;
-
   // Called just before this Command runs the first time to set your values
   public void initialize(double _P,double _I, double _D,
-                         double Cage_Limit,double Deadband,double MaxOutput) {
+                         double Cage_Limit, double Deadband, double MaxOutput) {
```

Comment:
Why even have a deadband and cage limit parameter in the constructor if you are not going to use it in the execution function?

Reply:
Wait the parameters get used for the upper/lower limits

Reply:
Yes, you set the limits. But both the flags that you use to check those limits are set to false. Thus they are never used. You should remove the limit initializations here and put them in the other constructor.